test/c/test_optimize_1q_decomposition.c

```diff
@@ -0,0 +1,315 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025.
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+#include "common.h"
+#include <complex.h>
+#include <math.h>
+#include <qiskit.h>
+#include <stdbool.h>
+#include <stddef.h>
+#include <stdint.h>
+#include <stdio.h>
+#include <string.h>
+
+// Helper headers
+QkTarget *get_u1_u2_u3_target(void);
+QkTarget *get_rz_rx_target(void);
+QkTarget *get_rz_sx_target(void);
+QkTarget *get_rz_ry_u_target(void);
+QkTarget *get_rz_ry_u_noerror_target(void);
+bool compare_gate_counts(QkOpCounts counts, char **gates, uint32_t *freq, int num_gates);
+
+/**
+ * Test running pass on chains of h gates.
+ *
+ * Transpile: 0:--[H]-[H]-[H]--
+ */
+int test_optimize_h_gates_inner(QkTarget *target, char **gates, uint32_t *freq, int num_gates) {
```

Comment:
Same comment about the naming here 🙂

Reply:
Addressed in 19f55a61197d9d6ac5b188778e4a2dd98b020b1c